pennylane_lightning/core/src/utils/cuda_utils/LinearAlg.hpp

```diff
@@ -80,6 +80,35 @@ class CublasCaller {
     cublasHandle_t handle;
 };
 
+/**
+ * @brief cuBLAS backed matrix-matrix multiply for GPU data.
+ *
+ * @tparam T Complex data-type. Accepts cuFloatComplex and cuDoubleComplex
+ * @param A Device data pointer of matrix A.
+ * @param B Device data pointer of matrix B.
+ * @param C Device data pointer of matrix C.
+ * @param m Row size of the matrix A.
+ * @param n Column size of the matrix B.
+ * @param k Column size of the matrix A and row size of the matrix B.
+ * @param dev_id the device on which the function should be executed.
+ * @param stream_id the CUDA stream on which the operation should be executed.
+ * @param cublas the CublasCaller object that manages the cuBLAS handle.
+ */
+template <class T = cuDoubleComplex, class DevTypeID = int>
+inline void GEMM_CUDA_device(T *A, T *B, T *C, const int m, const int k,
+                             const int n, DevTypeID dev_id,
+                             cudaStream_t stream_id,
+                             const CublasCaller &cublas) {
+    const T alpha{1.0, 0.0};
+    const T beta{0.0, 0.0};
+    if constexpr (std::is_same_v<T, cuFloatComplex>) {
+        cublas.call(cublasCgemm, dev_id, stream_id, CUBLAS_OP_N, CUBLAS_OP_N, m,
```

Comment:
You may want to use the the cuBlas handle to avoid `cublas.call`s. Something like:
```suggestion
inline void GEMM_CUDA_device(cublasHandle_t handle, T *A, T *B, T *C, const int m, const int k,
                             const int n, DevTypeID dev_id,
                             cudaStream_t stream_id,
                             const CublasCaller &cublas) {
    const T alpha{1.0, 0.0};
    const T beta{0.0, 0.0};
    if constexpr (std::is_same_v<T, cuFloatComplex>) {
        cublasCgemm(handle, dev_id, stream_id, CUBLAS_OP_N, CUBLAS_OP_N, m,
```

Reply:
The implementation was from nvidia. The `CublasCaller` object wraps `cublasHandles` to avoid explicitly passing cublasHandle around. To make it aligned with the rest of code base, I would like to keep it as it is for now and refactor it later if needed.